rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/action/TbCopyAttributesToEntityViewNodeTest.java

```diff
@@ -0,0 +1,287 @@
+import org.thingsboard.server.common.data.EntityView;
+import org.thingsboard.server.common.data.id.DeviceId;
+import org.thingsboard.server.common.data.id.EntityViewId;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.msg.TbMsgType;
+import org.thingsboard.server.common.data.objects.AttributesEntityView;
+import org.thingsboard.server.common.data.objects.TelemetryEntityView;
+import org.thingsboard.server.common.msg.TbMsg;
+import org.thingsboard.server.common.msg.TbMsgMetaData;
+import org.thingsboard.server.dao.entityview.EntityViewService;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.refEq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.thingsboard.server.common.data.msg.TbMsgType.ACTIVITY_EVENT;
+import static org.thingsboard.server.common.data.msg.TbMsgType.ATTRIBUTES_DELETED;
+import static org.thingsboard.server.common.data.msg.TbMsgType.ATTRIBUTES_UPDATED;
+import static org.thingsboard.server.common.data.msg.TbMsgType.INACTIVITY_EVENT;
+import static org.thingsboard.server.common.data.msg.TbMsgType.POST_ATTRIBUTES_REQUEST;
+
+@ExtendWith(MockitoExtension.class)
+public class TbCopyAttributesToEntityViewNodeTest {
+
+    private final TenantId TENANT_ID = new TenantId(UUID.fromString("9fdb1f05-dc66-4960-9263-ae195f1b4533"));
+    private final DeviceId DEVICE_ID = new DeviceId(UUID.fromString("1d453dc9-9333-476a-a51f-093cf2176e59"));
+
+    private TbCopyAttributesToEntityViewNode node;
+    private EmptyNodeConfiguration config;
+
+    @Mock
+    private TbContext ctxMock;
+    @Mock
+    private EntityViewService entityViewServiceMock;
+    @Mock
+    private RuleEngineTelemetryService telemetryServiceMock;
+
+    @BeforeEach
+    void setUp() throws TbNodeException {
+        node = new TbCopyAttributesToEntityViewNode();
+        config = new EmptyNodeConfiguration().defaultConfiguration();
+        var configuration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        node.init(ctxMock, configuration);
+    }
+
+    @Test
+    public void givenExistingAttributes_whenOnMsg_thenCopyAttributesToView() {
+        EntityViewId entityViewId = EntityViewId.fromString("a2109747-d1f4-475a-baaa-55f5d4897ad8");
+        EntityView entityView = new EntityView(entityViewId);
+        entityView.setStartTimeMs(Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+        entityView.setEndTimeMs(Instant.now().plus(1, ChronoUnit.DAYS).toEpochMilli());
```

Comment:
please make Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli() and Instant.now().plus(1, ChronoUnit.DAYS).toEpochMilli() final fields and reuse them in each test.

Reply:
moreover, consider to move creation of entity view to some method and reuse it each test.